### PR TITLE
VTOL params meta data

### PR DESCRIPTION
**Pull request by sanderux**
Augments https://github.com/PX4/Firmware/issues/3879
Please review unit names and enum values

### CODE REVIEW

File: src/modules/vtol_att_control/vtol_att_control_params.c [hunk 1]
```diff
@@ -49,7 +54,11 @@ PARAM_DEFINE_INT32(VT_MOT_COUNT, 0);
 /**
  * Idle speed of VTOL when in multicopter mode
  *
+ * @unit pulse-width
  * @min 900
+ * @max 2000
+ * @increment 1
+ * @deciml 0
```
Comment (LorenzMeier) [inline src/modules/vtol_att_control/vtol_att_control_params.c]: Typo, this broke Travis.

**Comment by LorenzMeier** [comment]
@sanderux Please register with Travis CI so you're getting email updates on build failures. You need to fix at least one typo here.

**Comment by sanderux** [comment]
Will do

**Comment by sanderux** [comment]
@LorenzMeier passed now, had to run travis twice

File: src/modules/vtol_att_control/vtol_att_control_params.c [hunk 2]
```diff
@@ -143,8 +180,12 @@ PARAM_DEFINE_FLOAT(VT_ARSP_LP_GAIN, 0.3f);
 /**
  * VTOL Type (Tailsitter=0, Tiltrotor=1, Standard=2)
  *
+ * @unit enum
+ * @value 0 Tailsitter 1 Tiltrotor 2 Standard
```
Comment (LorenzMeier) [inline src/modules/vtol_att_control/vtol_att_control_params.c]: Each value entry has to be on one line. You can't have multiple on a single line. This is structured markup.

File: src/modules/vtol_att_control/vtol_att_control_params.c [hunk 3]
```diff
@@ -248,8 +316,11 @@ PARAM_DEFINE_INT32(VT_WV_LTR_EN, 0);
  *
  * Time in seconds after which transition will be cancelled. Disabled if set to 0.
  *
- * @min 0.0
- * @max 30.0
+ * @unit seconds
+ * @min 0.00
+ * @max 30.00
+ * @increment 1
```
Comment (LorenzMeier) [inline src/modules/vtol_att_control/vtol_att_control_params.c]: increment 1 second and two decimals makes no sense. Please change the increment to a value that makes sense WRT the decimal places.

File: src/modules/vtol_att_control/vtol_att_control_params.c [hunk 4]
```diff
@@ -216,8 +275,11 @@ PARAM_DEFINE_INT32(VT_OPT_RECOV_EN, 0);
 /**
  * Enable weather-vane mode landings for missions
  *
+ * @unit bool
  * @min 0
  * @max 1
+ * @increment 1
```
Comment (LorenzMeier) [inline src/modules/vtol_att_control/vtol_att_control_params.c]: This should not have an increment, but two @value lines.

File: src/modules/vtol_att_control/vtol_att_control_params.c [hunk 5]
```diff
@@ -207,8 +263,11 @@ PARAM_DEFINE_FLOAT(VT_ARSP_TRANS, 10.0f);
 /**
  * Enable optimal recovery strategy for pitch-weak tailsitters
  *
+ * @unit bool
```
Comment (LorenzMeier) [inline src/modules/vtol_att_control/vtol_att_control_params.c]: Bool is not a unit, its a data structure

File: src/modules/vtol_att_control/vtol_att_control_params.c [hunk 6]
```diff
@@ -207,8 +263,11 @@ PARAM_DEFINE_FLOAT(VT_ARSP_TRANS, 10.0f);
...
  * @min 0
  * @max 1
+ * @increment 1
```
Comment (LorenzMeier) [inline src/modules/vtol_att_control/vtol_att_control_params.c]: Should not have increment, but should have two ```@value``` tags.

File: src/modules/vtol_att_control/vtol_att_control_params.c [hunk 7]
```diff
@@ -90,8 +111,11 @@ PARAM_DEFINE_FLOAT(VT_MC_ARSPD_TRIM, 10.0f);
  * If set to one this parameter will cause permanent attitude stabilization in fw mode.
  * This parameter has been introduced for pure convenience sake.
  *
+ * @unit bool
```
Comment (LorenzMeier) [inline src/modules/vtol_att_control/vtol_att_control_params.c]: Not a unit, again missing ```@value``` tags.

**Comment by LorenzMeier** [comment]
@sanderux The issues I raised are still open here.

**Comment by sanderux** [comment]
ping @LorenzMeier

**Comment by LorenzMeier** [comment]
Looking good, will merge as it passes.

**Pull request closed without merging**